### PR TITLE
Respond to timeouts with HTTP 408

### LINKED ISSUE #39: read timeout error
If the client takes too long to complete sending data and `readTimeout` value is exceeded, vulcand returns HTTP 400 and JSON body with `{"error":"Bad Request"}`

A more appropriate error might be HTTP 408  `{"error":"Request Timeout"}`

Relates to #35

### PR DESCRIPTION
**Pull request by porjo**
I've updated `proxy_test.go` but note that the test was failing even before I modified it. 

Fixes mailgun/vulcand#39

### CODE REVIEW

File: proxy.go [hunk 1]
```diff
@@ -27,13 +27,19 @@ type Options struct {
...
+// Accepts requests, round trips it to the endpoint, and writes back the response.
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Record the request body so we can replay it on errors.
 	body, err := netutils.NewBodyBuffer(r.Body)
 	if err != nil || body == nil {
 		log.Errorf("Request read error %s", err)
-		p.replyError(errors.FromStatus(http.StatusBadRequest), w, r)
+		// "i/o timeout" message is generated by net pkg in response
+		// to, for example, http.Server.ReadTimeout being exeeded.
+		if err.Error() == "i/o timeout" {
```
Comment (klizhentas) [inline proxy.go]: Can you use a more robust way to detect the timeout error, like this:

```
if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
    p.replyError(errors.FromStatus(http.StatusRequestTimeout), w, r)
}
```

Comment (klizhentas) [inline proxy.go]: Otherwise looks good

Comment (klizhentas) [inline proxy.go]: Also note that by the condition body can be nil, so just err.Error() can crash

**Comment by porjo** [comment]
Yes, your suggestion is much better! Fixed

**Pull request merged**